### PR TITLE
Add support for soft sector skew

### PR DESCRIPTION
**Pull request by davidgiven**
Some platforms, like the Apple II and CP/M, tend to use soft sector skew where the physical sectors get mapped to a logical sector via a table which isn't on the disk. FluxEngine needs to be told what that table is before it can do anything about it.

### CODE REVIEW

File: src/formats/appledos.textpb
Comment (jepler) [inline src/formats/appledos.textpb:6]: This is either correct, or exactly backwards.

File: lib/mapper.proto
Comment (davidgiven) [reply lib/mapper.proto:8]: If there are no mappings at all (i.e. an empty trackdata) then it's equivalent to the identity mapping. If there is a mapping specified and a sector isn't found in it, you'll get an runtime error --- _after_ the read happens, unfortunately.

File: lib/mapper.proto
Comment (davidgiven) [reply lib/mapper.proto:14]: They are merged top to bottom, in the same way that happens elsewhere. Honestly now I think of it I don't think there's a lot of value here, but it's the same code as used everywhere else...

File: src/formats/appledos.textpb
Comment (davidgiven) [reply src/formats/appledos.textpb:6]: Precisely.

**Review by jepler — APPROVED** [review]
Should be good to go after the merge conflicts are resolved.

**Comment by jepler** [comment]
It's backwards. Here's a mapping that worked for me:
```c
comment: 'Applies standard AppleDOS soft sector skew to the Apple II format'
is_extension: true

sector_mapping {
	trackdata {
		mapping { physical: 0  logical: 0  }
		mapping { physical: 1  logical: 7  }
		mapping { physical: 2  logical: 14 }
		mapping { physical: 3  logical: 6  }
		mapping { physical: 4  logical: 13 }
		mapping { physical: 5  logical: 5  }
		mapping { physical: 6  logical: 12 }
		mapping { physical: 7  logical: 4  }
		mapping { physical: 8  logical: 11 }
		mapping { physical: 9  logical: 3  }
		mapping { physical: 10 logical: 10 }
		mapping { physical: 11 logical: 2  }
		mapping { physical: 12 logical: 9  }
		mapping { physical: 13 logical: 1  }
		mapping { physical: 14 logical: 8  }
		mapping { physical: 15 logical: 15 }
	}
}
```
This, together with #489, let me get a complete working Apple DOS 3.3 image that is usable in mame.

Note that for best use in emulators, it would be nice to be able to use the appledos sector mapping when the `.do` extension is specified and the prodos sector mapping when the `.po` extension is specified. Is this feasible to implement?

I've prepared a pull request that updates the mapping, which I see you grabbed already while I was writing this up. yay. Thank you!

**Comment by davidgiven** [comment]
Thanks for doing this!

Changing the mapping by extension would be annoying --- not hard, but it'd need two bespoke imagereader/writer subclasses which automatically apply the mapping files. Doable but annoying...

**Pull request merged**
